packages/peregrine/lib/talons/MagentoRoute/useMagentoRoute.js

```diff
@@ -0,0 +1,45 @@
+import { useEffect, useState } from 'react';
+import { useLocation } from 'react-router-dom';
+import { useApolloClient } from '@apollo/react-hooks';
+
+import addToCache from './addToCache';
+import getRouteComponent from './getRouteComponent';
+
+const CLIENT_PATHS = new Set().add('/search.html');
+const IS_LOADING = { isLoading: true };
+
+export const useMagentoRoute = () => {
+    const [componentMap, setComponentMap] = useState(new Map());
+    const { apiBase } = useApolloClient();
+    const { pathname } = useLocation();
+
+    const routeData = componentMap.get(pathname);
+
+    // ask Magento for a RootComponent that matches the current pathname
+    useEffect(() => {
+        const isKnown = componentMap.has(pathname);
+        const isClientOnly = CLIENT_PATHS.has(pathname);
+
+        // avoid asking if we already know the answer
+        if (!isKnown && !isClientOnly) {
```

Comment:
In the current implementation, there's one more condition here, which is when [the known component has a certain `id`](https://github.com/magento/pwa-studio/blob/develop/packages/peregrine/lib/Router/magentoRouteHandler.js#L61). Is it still relevant?

Reply:
Fixed in [5882424](https://github.com/magento/pwa-studio/pull/1966/commits/5882424c9c17d2d66f1b8158e12405a26ab2a29a).